Sources/MapboxMaps/Foundation/Camera/CameraTransition.swift

```diff
@@ -0,0 +1,113 @@
+    public var shouldOptimizeBearingPath: Bool = true
+
+    /// Represents a change to the pitch of the map.
+    public var pitch: Change<CGFloat>
+
+    /// Generic struct used to represent a change in a value from a starting point (i.e. `fromValue`) to an end point (i.e. `toValue`).
+    public struct Change<T> {
+        public var fromValue: T
+        public var toValue: T?
+
+        init(fromValue: T, toValue: T? = nil) {
+            self.fromValue = fromValue
+            self.toValue = toValue
+        }
+    }
+
+    internal init(with renderedCameraOptions: CameraOptions, initialAnchor: CGPoint) {
+
+        guard let renderedCenter = renderedCameraOptions.center,
+              let renderedZoom = renderedCameraOptions.zoom,
+              let renderedPadding = renderedCameraOptions.padding,
+              let renderedPitch = renderedCameraOptions.pitch,
+              let renderedBearing = renderedCameraOptions.bearing else {
+            fatalError("Values in rendered CameraOptions cannot be nil")
+        }
+
+        center  = .init(fromValue: renderedCenter)
+        zoom    = .init(fromValue: renderedZoom)
+        padding = .init(fromValue: renderedPadding)
+        pitch   = .init(fromValue: renderedPitch)
+        bearing = .init(fromValue: renderedBearing)
+        anchor  = .init(fromValue: initialAnchor)
+    }
+
+    internal var toCameraOptions: CameraOptions {
+        return CameraOptions(center: center.toValue,
+                             padding: padding.toValue,
+                             anchor: anchor.toValue,
+                             zoom: zoom.toValue,
+                             bearing: shouldOptimizeBearingPath ? Self.optimizeBearing(startBearing: bearing.fromValue,
+                                                                                       endBearing: bearing.toValue)
+                                                                :  bearing.toValue,
+                             pitch: pitch.toValue)
+    }
+
+    internal var fromCameraOptions: CameraOptions {
+        return CameraOptions(center: center.fromValue,
+                             padding: padding.fromValue,
+                             anchor: anchor.fromValue,
+                             zoom: zoom.fromValue,
+                             bearing: bearing.fromValue,
+                             pitch: pitch.fromValue)
+
+    }
+
+    /// This function optimizes the bearing for set camera so that it is taking the shortest path.
+    /// - Parameters:
+    ///   - startBearing: The current or start bearing of the map viewport.
+    ///   - endBearing: The bearing of where the map viewport should end at.
+    /// - Returns: A `CLLocationDirection` that represents the correct final bearing accounting for positive and negatives.
+    internal static func optimizeBearing(startBearing: CLLocationDirection?, endBearing: CLLocationDirection?) -> CLLocationDirection? {
```

Comment:
does this need to be static? if it were non-static, it could just be a computed property named `optimizedBearing`

Reply:
or `optimizedBearingToValue`

Reply:
doesn't have to be static no.. I can make it a computed property.